TestRegisterCRD is flaky

When the tests ran for https://github.com/kubernetes-sigs/kustomize/pull/427 they contained the following failure. This is an unrelated codepath to the PR and was resoved by closing and reopening the PR. On the second run this test passed.

```
--- FAIL: TestRegisterCRD (0.00s)
	crds_test.go:185: expected
		 &{[] [] [] [] [noGroup_v1beta1_Bee noGroup_v1_Secret] []}
		 but got
		 &{[] [] [] [] [noGroup_v1_Secret noGroup_v1beta1_Bee] []}
```

The [DeepEquals assertion](https://github.com/kubernetes-sigs/kustomize/blob/7cf5f8caaef26af3721248d99a1f6d2ec44e7bc4/pkg/crds/crds_test.go#L184) is failing due to order of the []transformerconfig.ReferencePathConfig slice.

```
	if !reflect.DeepEqual(pathconfig, expected) {
		t.Fatalf("expected\n %v\n but got\n %v\n", expected, pathconfig)
	}
```

The slice is populated by [a code path that iterates over a map](https://github.com/kubernetes-sigs/kustomize/blob/7cf5f8caaef26af3721248d99a1f6d2ec44e7bc4/pkg/crds/crds.go#L107) that has inheritly unstable keys. There are several possible solutions here ( load in a stable sorted order, perform assertions that ignore sort order, ... ). So I thought I'd open this to discuss rather than just fix.

Reference: [Build 826](https://travis-ci.org/kubernetes-sigs/kustomize/builds/438455785) and [Build 827](https://travis-ci.org/kubernetes-sigs/kustomize/builds/438456974?utm_source=github_status&utm_medium=notification)

--- a/pkg/crds/crds_test.go
+++ b/pkg/crds/crds_test.go
@@ -18,6 +18,7 @@ package crds
 
 import (
 	"reflect"
+	"sort"
 	"testing"
 
 	"sigs.k8s.io/kustomize/pkg/gvk"
@@ -180,6 +181,14 @@ func TestRegisterCRD(t *testing.T) {
 	ldr := makeLoader(t)
 
 	pathconfig, _ := registerCRD(ldr, "/testpath/crd.json")
+
+	sort.Slice(expected.NameReference[:], func(i, j int) bool {
+		return expected.NameReference[i].Gvk.String() < expected.NameReference[j].Gvk.String()
+	})
+
+	sort.Slice(pathconfig.NameReference[:], func(i, j int) bool {
+		return pathconfig.NameReference[i].Gvk.String() < pathconfig.NameReference[j].Gvk.String()
+	})
 
 	if !reflect.DeepEqual(pathconfig, expected) {
 		t.Fatalf("expected\n %v\n but got\n %v\n", expected, pathconfig)

